service/src/lib.rs

```diff
@@ -236,8 +236,14 @@ where
 
     let role = config.role.clone();
     let force_authoring = config.force_authoring;
-    let backoff_authoring_blocks =
-        Some(sc_consensus_slots::BackoffAuthoringOnFinalizedHeadLagging::default());
+    let backoff_authoring_blocks = {
+        let mut backoff = sc_consensus_slots::BackoffAuthoringOnFinalizedHeadLagging::default();
+        // the default max interval is 100 slots, but since we use small session duration
+        // (50 slots) we want to make sure we will never backoff authoring blocks for
+        // longer than one epoch duration.
+        backoff.max_interval = 10;
```

Comment:
We can also tune `backoff.unfinalized_slack` which determines how many unfinalized blocks until the block production backoff triggers, by default it is 50 blocks.

Reply:
@andresilva `max_interval` means no matter how far the finality is lagging, the node will try producing blocks less than `max_interval` blocks, is that right? Can it be set at a lower value than 10 because that will make the network keep producing blocks so that we can have a chance to send the sudo extrinsic `note_stalled`, the network will pack this fixing extrinsic faster?

I have a sense that making `unfinalized_slack` smaller can decrease the impact of bad things since the number of unfinalized blocks are no more than 3 when everything is in a normal state, no?

Reply:
> @andresilva `max_interval` means no matter how far the finality is lagging, the node will try producing blocks less than `max_interval` blocks, is that right? Can it be set at a lower value than 10 because that will make the network keep producing blocks so that we can have a chance to send the sudo extrinsic `note_stalled`, the network will pack this fixing extrinsic faster?

We can also disable the backoff altogether if you think this is preferrable.

> I have a sense that making `unfinalized_slack` smaller can decrease the impact of bad things since the number of unfinalized blocks are no more than 3 when everything is in a normal state, no?

Reducing `unfinalized_slack` means that the backoff starts happening earlier, e.g. if `unfinalized_slack` is 50 there is no backoff until the finality lag is >50, and then the nodes will start progressively authoring blocks slower until the point where they only author every `max_interval`.

Reply:
>We can also disable the backoff altogether if you think this is preferrable.

Hmm, correct me if I'm wrong, if we disable the backoff, the nodes will just keep authoring blocks without increasing the finality, and the node simply ignores the finality lagging. Once we observe the issue happened, basically we can always trigger `note_stalled` to recover the finality no matter how far the lagging is?

Reply:
Yes, with the backoff disabled the nodes will just keep authoring blocks forever on-time no matter how much finality lag there is. And yes, `note_stalled` allows you to forcefully change to a new grandpa set no matter how long finality is lagging.

Reply:
I can update the PR to disable it if you prefer this.

Reply:
>I can update the PR to disable it if you prefer this.

I'm for that. Since we always have a way to fix the finality, that will make the network always alive.

Reply:
@andresilva Another question, is the backoff strategy changeable after the network has been already launched?